src/Messaging/MessageDataAssertion.php

```diff
@@ -102,15 +90,12 @@ public static function assertMetadata($metadata)
         }
     }
 
-    /**
-     * @param \DateTimeInterface $createdAt
-     */
-    public static function assertCreatedAt($createdAt)
+    public static function assertCreatedAt($createdAt): void
     {
         Assertion::isInstanceOf($createdAt, \DateTimeInterface::class, sprintf(
```

Comment:
I´m fully agree with immutable messages at all. It seems that this was a bug in MessageDataAssertion and should also defined as it. Maybe with an extra reference in release changeset. This BC break is stronger than "we have added some type hints". Right?